From 1c44915a462e393afe990a42c87dab151b49f832 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Fri, 19 Apr 2019 13:07:45 +0300
Subject: [PATCH] [SW-0] pll: Gaudi introduce dip switch skip

If set then do not initialize any of the PLLs

Change-Id: I80a717f84e2e731375ddb7fa3431976a62fa04d4
---

diff --git a/gaudi/gaudi.c b/gaudi/gaudi.c
index e609ddb..05db649 100644
--- a/gaudi/gaudi.c
+++ b/gaudi/gaudi.c
@@ -127,6 +127,10 @@
 int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg)
 {
 	int pll_num = ARRAY_SIZE(gaudi_pll_cfg);
+	int boot_starp = regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS);
+
+	if (boot_starp & SKIP_PLL_INIT)
+		return 0;
 
 	*hl_pll_cfg = gaudi_pll_cfg;
 
diff --git a/gaudi/gaudi.h b/gaudi/gaudi.h
index 37564b4..8af8949 100644
--- a/gaudi/gaudi.h
+++ b/gaudi/gaudi.h
@@ -7,6 +7,8 @@
 
 /* Use DumpSel[11] to skip PCIe initialization */
 #define SKIP_PCIE_INIT BIT(19)
+/* Use DumpSel[9] to skip PLL initialization */
+#define SKIP_PLL_INIT BIT(17)
 
 #define mmDMA_W_PLL_BASE		0x487000
 #define mmIF_W_PLL_BASE			0x488000
